ctrlf.el

```diff
@@ -253,19 +334,15 @@ fails, return nil, but still move point."
   (ctrlf--clear-transient-overlays)
   (cl-block nil
     (let ((input (field-string (point-max))))
-      (if ctrlf--regexp-p
-          (condition-case e
-              (when (string-match-p input "")
-                ;; Let's just rule out zero-length matches entirely,
-                ;; they're not interesting and they make the
-                ;; implementation more complicated and slower.
-                (cl-return))
-            (invalid-regexp
-             (ctrlf--message "Invalid regexp: %s" (cadr e))
-             (cl-return)))
-        ;; Also exclude zero-length matches for literal searches.
-        (when (string-empty-p input)
-          (cl-return)))
+      (condition-case e
+          (when (string-match-p input "")
+            ;; Let's just rule out zero-length matches entirely,
+            ;; they're not interesting and they make the
+            ;; implementation more complicated and slower.
+            (cl-return))
+        (invalid-regexp
+         (ctrlf--message "Invalid regexp: %s" (cadr e))
```

Comment:
Currently this will happen even when we are doing literal search. Ideally we shouldn't check ctrlf--style here, but check the translated regexp is invalid or not, and I've not come up with a good way to do that.